tests/test_frame.py

```diff
@@ -486,6 +486,182 @@ def gethead(s: pd.Series, y: int) -> pd.Series:
 
     check(assert_type(df.apply(gethead, args=(4,)), pd.DataFrame), pd.DataFrame)
 
+    # Check scalar/series/list-like for default (None), and result type of expand, reduce, broadcast
+    check(
+        assert_type(df.apply(returns_scalar), pd.Series),
+        pd.Series,
+    )
+    check(
+        assert_type(df.apply(returns_series), pd.DataFrame),
+        pd.DataFrame,
+    )
+    check(
+        assert_type(df.apply(returns_listlike_of_3), pd.DataFrame),
+        pd.DataFrame,
+    )
+
+    # While this call works in reality, it errors in the type checker, because this should never be called
+    # It does not make sense to pass a result_type of "expand" to a scalar return
+    # check(
+    #     assert_type(
+    #         df.apply(returns_scalar, result_type="expand"), pd.DataFrame
+    #     ),
+    #     pd.DataFrame,
+    # )
```

Comment:
When I run the above, it returns `Series` so it seems that you typed it wrong when testing.  Can you put it back using `pd.Series` ?

Reply:
It may make no sense, but I tried it and it worked.  In other words, it doesn't do nothing.  It does something!  So not sure what to do about that!

Reply:
By nothing I mean that specifying `result_type="expand"` gives the same result as if you don't specify it (because it's not possible to expand a scalar).

The reason why it doesn't throw is because pandas has no way of knowing whether your function always returns a scalar or not, it has to assume that it's possible for some values to be scalars and some values to be list-like.

However, with annotations, we do know. Because a `Callable[[pd.Series], int]` can be assumed to always return an `int`.

I can certainly put it back, just felt that it might make sense to disallow it...

Reply:
I see your point, but we've taken the position that if the code works with pandas, we can't disallow it in the typing.  If someone is using `expand` with a `Scalar`, while it doesn't make sense, one could argue that our documentation doesn't say it is invalid.

Reply:
Fair enough, I will uncomment these test cases and adjust the overloads as needed